### PR TITLE
Upgrade SQLAlchemy

### PR DESCRIPTION
**Pull request by sea-kelp**
## Description of Changes
Upgrade SQLAlchemy (finally)! Most changes are adapted from https://github.com/lucyparsons/OpenOversight/commit/14bfc5b13656398c84c8a61744579ce4ade4e548

## Notes for Deployment
None!

## Screenshots (if appropriate)
N/A

## Tests and linting

 - [x] I have rebased my changes on `main`

 - [x] `just lint` passes

 - [x] `just test` passes

### CODE REVIEW

File: OpenOversight/tests/routes/test_image_tagging.py [hunk 1]
```diff
@@ -122,8 +122,9 @@ def test_ac_cannot_delete_tag_not_in_their_dept(mockdata, client, session):
         login_ac(client)
 
         tag = (
-            Face.query.join(Face.officer, aliased=True)
-            .except_(Face.query.filter(Face.officer.has(department_id=AC_DEPT)))
+            Face.query.join(Officer, Face.officer_id == Officer.id)
+            .join(Department, Officer.department_id == Department.id)
+            .filter(Department.id != AC_DEPT)
```
Comment (sea-kelp) [inline OpenOversight/tests/routes/test_image_tagging.py:127]: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#orm-query-join-aliased-true-from-joinpoint-removed

File: OpenOversight/tests/test_utils.py [hunk 2]
```diff
@@ -108,11 +108,12 @@ def test_filter_by_name(mockdata):
...
 
-def test_filters_do_not_exclude_officers_without_assignments(mockdata):
+def test_filters_do_not_exclude_officers_without_assignments(mockdata, db):
     department = OpenOversight.app.models.Department.query.first()
     officer = OpenOversight.app.models.Officer(
         first_name="Rachel", last_name="S", department=department, birth_year=1992
     )
+    db.session.add(officer)
     results = OpenOversight.app.utils.grab_officers({"name": "S", "dept": department})
     assert officer in results.all()
```
Comment (sea-kelp) [inline OpenOversight/tests/test_utils.py:118]: Idk how this test has been passing :thinking:

File: docker-compose.dev.yml [hunk 3]
```diff
@@ -49,6 +49,7 @@ services:
     environment:
       ENVIRONMENT: development
       FLASK_DEBUG: 1
+      SQLALCHEMY_WARN_20: 1
```
Comment (sea-kelp) [inline docker-compose.dev.yml:52]: Turn on SQLAlchemy 2.x deprecation warnings

Comment (AetherUnbound) [reply docker-compose.dev.yml:52]: Excellent!

**Review by AetherUnbound — APPROVED** [review]
Thanks so much for taking this on! I was able to build & test this locally and it worked great 🚀

**Pull request merged**